feat: ability to search in installed apidocs

emmanueltouzery opened the pull request:
i think this feature is useful. maybe it's a little crude, but it's better than not having the feature at all...

emmanueltouzery committed:
feat: ability to search in installed apidocs

luckasRanarison commented:
Intersting, I didn't know I needed this until now. But naming it `DevdocsGrep` would be better I think.

emmanueltouzery commented:
Sure, I can rename it a little later today! Why I wrote it was crude...

- it doesn't respect the setting to pipe though `glow`
-  it shows the internal filenames such as 1.md instead of the real keys such as.. String.reverse or whatever

But going all the way would be quite a lot more work, and possibly too slow to be useful. I guess this is good enough, and can be improved though time potentially.

emmanueltouzery commented:
modified the PR to now use the Grep terminology.

luckasRanarison commented:
I think displaying the raw markdown is okay in this case, using `glow` actually doesn't make sense. 
I found that we can hook in the `live_grep` builtin and change the display by passing a custom `entry_maker`. Having this with floating window would aso be nice but it seems that we can't directly change the default selection mapping, so that'd require some silly workaround. I'll take a closer look later but I think it can be improved.

luckasRanarison commented:
So I improved the result display format...
>     * it doesn't respect the setting to pipe though `glow`
I only realize it now, I thought you were talking about the previewer. The picker would just open the file in the current implementation.

luckasRanarison commented:
We could get the last preview buffer using `telescope.state` but the main problem is making it work with glow. We don't have enough information so we'd have to pass the entire buffer to glow, and you know what'd would happen with large files, I also don't know how we can implement jumping in this case.
Any thoughts on this?

luckasRanarison commented:
Glow output doesn't necessarily match the original one so jumping to the specified line won't always work. But this feature is still great and I just learned about `conceallevel`, with it set to 3 previewing raw markdown in neovim with treesitter is almost good as using glow. I'm working on some fixes now and I'll set it to 3 by default.

emmanueltouzery committed:
fix a crash when closing a picker with no matches

https://github.com/luckasRanarison/nvim-devdocs/pull/60
it was reverted in the original repo with https://github.com/luckasRanarison/nvim-devdocs/commit/fea272d185d7230fd1bfca304f73d06954032c70
but i still think this should be deleted and that fixes the close issue
open a picker, type text until there are no matches, close the picker
